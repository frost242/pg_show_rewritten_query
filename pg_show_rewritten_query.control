# pg_show_rewritten_query extension
comment = 'Print a query after rewrite stage'
default_version = '1.0.0'
module_pathname = '$libdir/pg_show_rewritten_query'
relocatable = true
