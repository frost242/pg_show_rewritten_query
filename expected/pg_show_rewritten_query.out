CREATE EXTENSION pg_show_rewritten_query;
CREATE TABLE t_query_def(id integer, dpt integer, nb numeric);
CREATE VIEW v_query_def AS SELECT dpt, sum(nb) AS sum_nb
FROM t_query_def WHERE dpt IS NOT NULL GROUP BY dpt;
SELECT pg_show_rewritten_query('SELECT * FROM v_query_def');
             pg_show_rewritten_query              
--------------------------------------------------
  SELECT dpt,                                    +
     sum_nb                                      +
    FROM ( SELECT t_query_def.dpt,               +
             sum(t_query_def.nb) AS sum_nb       +
            FROM t_query_def                     +
           WHERE t_query_def.dpt IS NOT NULL     +
           GROUP BY t_query_def.dpt) v_query_def;+
 
(1 row)

